cargo-dist/src/build/generic.rs

```diff
@@ -216,6 +217,16 @@ pub fn build_generic_target(
     // Check and process the binaries
     expected.process_bins(dist_graph, manifest)?;
 
+    // Record the build environment
+    let system_id = format!("build:local:{}", target.target_triple);
+    if let Some(info) = manifest.systems.get_mut(&system_id) {
+        // Not yet set; do that now
+        if matches!(info.build_environment, BuildEnvironment::Indeterminate) {
+            let build_environment = determine_build_environment(&target.target_triple)?;
+            info.build_environment = build_environment;
+        }
+    };
```

Comment:
I don't really get why you compute this in the middle of each build step -- it's a global property of the system we're running on. This could be computed in compute_build_steps or anything more universal/shared, right? (mostly don't want code duplication here)

Reply:
Yes, that's also true. I sort of automatically associated "build environment properties" with the build process itself, but they don't have to be coupled *that* closely. (I guess it's possible for the build to mutate the environment we capture, but it's probably *very uncommon* for the build itself to obsolete the environment we calculate prebuild.)